ranger/gui/widgets/taskview.py

```diff
@@ -53,9 +54,11 @@ def draw(self):
                         clr.append('selected')
 
                     descr = obj.get_description()
-                    if obj.progressbar_supported and obj.percent >= 0 and obj.percent <= 100:
-                        self.addstr(y, 0, "%3.2f%% - %s" % (obj.percent, descr), self.wid)
-                        wid = int((self.wid / 100) * obj.percent)
+                    if obj.progressbar_supported and 0 <= obj.progress.percent <= 100:
+                        rate = human_readable(obj.progress.average_rate, '') + "/s"
+                        status = '{:3.2f}% {:>7} - {:s}'.format(obj.progress.percent, rate, descr)
+                        self.addstr(y, 0, status, self.wid)
+                        wid = int((self.wid / 100) * obj.progress.percent)
```

Comment:
Not your change, but when reading `wid` I think of `w_id`, or more precisely "window ID" and not `width`. That's a quite unfortunate shortening, but it's also used in the class attribute.

Reply:
Ok, I added that to my TODO list. I might rename this before I try expanding the progress to multiple rows (if I ever justify doing that).